test/node-api.js

```diff
@@ -6,11 +6,188 @@ chai.use(codeEquality);
 
 const { expect } = chai;
 
-const p = new Preprocessor();
+describe("parse", function() {
+  it("basic example", function() {
+    let p = new Preprocessor();
+    let output = p.parse("<template>Hello!</template>");
 
-describe("something", function () {
-  it("works for a basic example", function () {
-    let output = p.process("<template>Hi</template>");
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 0,
+          end: 27
+        },
+        contentRange: {
+          start: 10,
+          end: 16
+        },
+        startRange: {
+          end: 10,
+          start: 0
+        },
+        endRange: {
+          start: 16,
+          end: 27
+        }
+      },
+    ]);
+  });
+
+  it("expression position", function() {
+    let p = new Preprocessor();
+    let output = p.parse("const tpl = <template>Hello!</template>");
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 12,
+          end: 39
+        },
+        contentRange: {
+          start: 22,
+          end: 28
+        },
+        startRange: {
+          start: 12,
+          end: 22
+        },
+        endRange: {
+          start: 28,
+          end: 39
+        }
+      },
+    ]);
+  });
+
+  it("inside class body", function() {
+    let p = new Preprocessor();
+    let output = p.parse(`
+      class A {
+        <template>Hello!</template>
+      }
+    `);
+
+    expect(output).to.eql([
+      {
+        type: "template-tag",
+        tagName: "template",
+        contents: "Hello!",
+        range: {
+          start: 25,
+          end: 52
+        },
+        contentRange: {
+          start: 35,
+          end: 41
+        },
+        startRange: {
+          start: 25,
+          end: 35
+        },
+        endRange: {
+          start: 41,
+          end: 52
+        }
+      },
+    ]);
+  });
+
+  it("preceded by a slash character", function() {
+    let p = new Preprocessor();
+    // What is this testing?
```

Comment:
the ember-templace-imports parser historically broke every time you had a `/` before `<template>`